src/pa02/test_pa02.py

```diff
@@ -0,0 +1,201 @@
+        player = cs.Player(board)
+
+        for _ in range(50):
+            player.move()
+            assert 1 <= player.position
+
+    def test_start_of_chute(self):
+        """Test that the players position can´t be the start point of a
+        snake or a ladder, since the player is supposed to interact with
+        these. """
+
+        board = cs.Board()
+        player = cs.ResilientPlayer(board)
+
+        for step in range(50):
+            player.move()
+            assert player.position not in board.snakes_and_ladders.keys()
+
+
+class TestLazyPlayer:
+    """Tests for LazyPlayer class
+    The following test, has the same function as those in Player class. The
+    only different is that the test for LazyPlayer. Therefore the follow the
+    same docstrings as in Player class, and I will not be repeated"""
+
+    def test_move(self):
+
+        board = cs.Board()
+        player = cs.LazyPlayer(board)
+        for _ in range(10):
+            player.move()
+
+        assert player.number_of_moves == 10
+
+    def test_position(self):
+
+        board = cs.Board()
+        player = cs.LazyPlayer(board)
+        start_pos = player.position
+        player.move()
+        end_pos = player.position
+
+        assert start_pos is not end_pos
+
+    def test_position_not_equal_one(self):
+
+        board = cs.Board()
+        player = cs.LazyPlayer(board)
+
+        for _ in range(50):
+            player.move()
+            assert 1 <= player.position
+
+    def test_start_of_chute(self):
+
+        board = cs.Board()
+        player = cs.ResilientPlayer(board)
+
+        for step in range(50):
+            player.move()
+            assert player.position not in board.snakes_and_ladders.keys()
+
+
+class TestResilientPlayer:
+    """Tests for ResilientPlayer class
+    The following test, has the same function as those in Player class. The
+    only different is that the test for ResilientPlayer. Therefore the
+    follow the same docstrings as in Player class, and I will not be
+    repeated"""
+
+    def test_move(self):
+
+        board = cs.Board()
+        player = cs.ResilientPlayer(board)
+        for _ in range(10):
+            player.move()
+
+        assert player.number_of_moves == 10
+
+    def test_position(self):
+
+        board = cs.Board()
+        player = cs.ResilientPlayer(board)
+        start_pos = player.position
+        player.move()
+        end_pos = player.position
+
+        assert start_pos is not end_pos
+
+    def test_position_not_equal_one(self):
+
+        board = cs.Board()
+        player = cs.ResilientPlayer(board)
+
+        for _ in range(50):
+            player.move()
+            assert 1 <= player.position
+
+    def test_start_of_chute(self):
+
+        board = cs.Board()
+        player = cs.ResilientPlayer(board)
+
+        for step in range(50):
+            player.move()
+            assert player.position not in board.snakes_and_ladders.keys()
+
+
+class TestSimulation:
```

Comment:
This is not how we handle randomness in real-life. And you used the result from your code to create tests, which makes these tests meaningless.

Reply:
There are other things you can test in Simulation:
- Single_game: make the board with ladders: 1 --> 90, 2-->90, 3 --> 90, 4 --> 90, 5 -->90, 6-->90. Then the first player will always win.
- Results: length of the return list should be the same with number of simulation. The first element in each tuple in that list should always greater than 0, and the second element should belongs to the list of different players.
- winners: the sum of all values in the dict should be equal with number of simulation.
- duration: the sum of the length of each value in the dict should be equal with number of simulation.